modules/extreme_year.py

```diff
@@ -0,0 +1,117 @@
+"""
+this module will return the highest and lowest temperatures of a
+given year along with the most humid day.
+"""
+import glob
+
+
+def read_and_calculate_extremes(pattern, path):
+    """
+    this function reads all the files at the given path which contain the given pattern
+    from each file that matches the pattern finds the following:
+    1. max highest temperature
+    2. max lowest temperature
+    3. max humidity
+    :param pattern: a string which contains a year e.g '2006', '2007'
+    :param path: a string which contains path to a directory containing weather files
+    :return: a 3x2 list maximums where
+    maximums[0] = [max_highest_temperature, date]
+    maximums[1] = [max_lowest_temperature, date]
+    maximums[2] = [max_humidity, date]
+    date here is a string like: '2002-4-1', '2004-3-14'
+    """
+    files = glob.glob(path + f"*{pattern}*")
+    if not files:
+        print(
+            f"We don't have information regarding the "
+            f"year {pattern} in the given path"
+        )
+        return -1
+    # maximums[0] contains highest temperature and day
+    # maximums[1] contains lowest temperature and day
+    # maximums[2] contains most humidity and day
+    maximums = [[0, ""], [0, ""], [0, ""]]
```

Comment:
Defining them separately would make more sense, With proper naming for each.

Reply:
done